Assets/Mirage/Components/Visibility/SpatialHash/SpatialHashVisibility.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using Mirage.Logging;
+using UnityEngine;
+
+namespace Mirage.Visibility.SpatialHash
+{
+    public class SpatialHashVisibility : NetworkVisibility
+    {
+        static readonly ILogger logger = LogFactory.GetLogger(typeof(SpatialHashVisibility));
+
+        [Tooltip("How many grid away the player can be to see this object. Real distance is this mutlipled by SpatialHashSystem")]
+        public int GridVisibleRange = 1;
+
+        [ReadOnlyInspector]
+        public SpatialHashSystem System;
+
+        /// <param name="player">Network connection of a player.</param>
+        /// <returns>True if the player can see this object.</returns>
+        public override bool OnCheckObserver(INetworkPlayer player)
+        {
+            ThrowIfNoSystem();
```

Comment:
should return and wait for system instead of throw. System can rebuild existing NI when it starts.

This is to better deal with spawning objects inside `server.Started`, the system might not be first in even list